sentry/src/routes/channel.rs

```diff
@@ -846,4 +973,156 @@ mod test {
+            channel: channel_context.context,
+            deposit: Deposit {
+                total: UnifiedNum::from_u64(1000),
+                still_on_create2: UnifiedNum::from_u64(0),
+            },
+        };
+
+        // Add accounting for earner -> 100
+        update_accounting(
+            app.pool.clone(),
+            channel_context.context.id(),
+            auth.uid.to_address(),
+            Side::Earner,
+            UnifiedNum::from_u64(100),
+        )
+        .await
+        .expect("should update");
+        // available for return is now 600
+
+        // Updating spendable so that we have a value for total_deposited
+        update_spendable(app.pool.clone(), &spendable)
+            .await
+            .expect("Should update spendable");
+
+        // Test with no body
+        let res =
+            channel_payout(build_request(&channel_context, UnifiedMap::default()), &app).await;
+        assert!(
+            matches!(res, Err(ResponseError::FailedValidation(..))),
+            "Failed validation because request body is empty"
+        );
+        // make a normal request and get accounting to see if its as expected
+
+        let mut payouts = UnifiedMap::default();
+        payouts.insert(*PUBLISHER, UnifiedNum::from_u64(500));
+
+        let res = channel_payout(build_request(&channel_context, payouts.clone()), &app).await;
+        assert!(
+            res.is_ok(),
+            "Request with JSON body with one address and no errors triggered on purpose"
+        );
+        let accounting = get_accounting(
+            app.pool.clone(),
+            channel_context.context.id(),
+            *PUBLISHER,
+            Side::Earner,
+        )
+        .await
+        .expect("should get accounting")
+        .expect("Should be some");
+        assert_eq!(
+            accounting.amount,
+            UnifiedNum::from_u64(500),
+            "Accounting is updated to reflect the publisher's earnings"
+        );
+
+        // make a request where total - spent + earned will be a negative balance resulting in an error
+        update_accounting(
+            app.pool.clone(),
+            channel_context.context.id(),
+            auth.uid.to_address(),
+            Side::Spender,
+            UnifiedNum::from_u64(1000),
+        )
+        .await
+        .expect("should update"); // total spent: 500 + 1000
+
+        let res =
+            channel_payout(build_request(&channel_context, UnifiedMap::default()), &app).await;
+        assert!(
+            matches!(res, Err(ResponseError::FailedValidation(..))),
+            "Failed validation because available_for_return is negative"
+        );
+
+        // make a request where "total_to_pay" will exceed available
+        payouts.insert(*CREATOR, UnifiedNum::from_u64(1000));
+
+        let res = channel_payout(build_request(&channel_context, payouts.clone()), &app).await;
+        assert!(
+            matches!(res, Err(ResponseError::FailedValidation(..))),
```

Comment:
It would be better to have assert_eq and also assert the message returned, as it might fail for other reasons with FailedValidatoin.